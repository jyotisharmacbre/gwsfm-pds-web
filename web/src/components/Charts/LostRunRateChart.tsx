import React from 'react';

import { Doughnut } from 'react-chartjs-2';

const dataLost = {
  datasets: [
    {
      data: [4, 2, 7, 9, 23, 2],
      backgroundColor: [
        // '#ced0fe',
        // '#7e83fe','#704ba4', '#4cbd7f',
        // '#f3ca55'
        '#CEF1A3',
        '#526061',
        '#FBD55B',
        '#935FD6',
        '#EA8032',
        '#3DAAE9'
      ],
      label: 'My dataset' // for legend
    }
  ],
  labels: [
    'Initial Customer Enquiry',
    'J&A',
    'Bid Submitted',
    'Order Received',
    'In Progress',
    'Completed'
    //ICE, J&A, Bid Submitted, Order Received, In Progress and Completed
  ]
};

const optionshideLabels = {
  showTotalLabel: true,
  cutoutPercentage: 70,
  legend: {
    display: false
  } //, maintainAspectRatio:false
};

export default function LostRunRateChart(props: {
  ICE: number;
  JandA: number;
  BidSubmitted: number;
  OrderReceived: number;
  InProgress: number;
  Completed: number;
}) {
  return <Doughnut data={dataLost} options={optionshideLabels} />;
}
