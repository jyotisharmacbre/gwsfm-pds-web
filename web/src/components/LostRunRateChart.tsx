import React from 'react';

import { Doughnut, Pie, Polar, Bubble, Line } from 'react-chartjs-2';
import Grid from '@material-ui/core/Grid';

  const dataLost = {
    datasets: [{
      data: [
        4,2,7,9,23,2
        
      ],
      backgroundColor: [
        // '#ced0fe',
        // '#7e83fe','#704ba4', '#4cbd7f',
        // '#f3ca55'
        '#CEF1A3',
        '#526061','#FBD55B', '#935FD6',
        '#EA8032','#3DAAE9'
       
      ],
      label: 'My dataset' // for legend
    }],
    labels: [
      'Initial Customer Enquiry',
      'J&A',
      'Bid Submitted','Order Received','In Progress','Completed'
      //ICE, J&A, Bid Submitted, Order Received, In Progress and Completed
    ]
  };
 
  const optionshideLabels={  
 
      legend: {
        display: false
      }//, maintainAspectRatio:false     
};

export default function LostRunRateChart(props: {ICE:number, JandA:number, BidSubmitted:number, OrderReceived:number, InProgress:number , Completed:number}){
    return (  <Doughnut   data={dataLost} options={optionshideLabels} /> 
             );
   
   
  }
  