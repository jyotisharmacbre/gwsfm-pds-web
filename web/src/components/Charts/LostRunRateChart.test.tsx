import React from 'react';
import LostRunRateChart from './LostRunRateChart';
import Enzyme, { mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() });

it('Lost Run Rate & OnHold Chart should render', () => {
  const component = shallow(
    <LostRunRateChart
      ICE={56}
      JandA={12}
      BidSubmitted={30}
      OrderReceived={20}
      InProgress={24}
      Completed={20}
    />
  );
  // let tree = component.toJSON();
  expect(component.getElements()).toMatchSnapshot();
});

it('Graph Grid exist', () => {
  const props = {
    ICE: 56,
    JandA: 12,
    BidSubmitted: 12,
    OrderReceived: 34,
    InProgress: 56,
    Completed: 45
  };
  const component = mount(<LostRunRateChart {...props} />);

  expect(component.find('#GridLostRunRateChart').exists);
});
