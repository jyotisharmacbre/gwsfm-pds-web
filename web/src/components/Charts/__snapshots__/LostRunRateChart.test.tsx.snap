// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lost Run Rate & OnHold Chart should render 1`] = `
Array [
  <Doughnut
    data={
      Object {
        "datasets": Array [
          Object {
            "backgroundColor": Array [
              "#CEF1A3",
              "#526061",
              "#FBD55B",
              "#935FD6",
              "#EA8032",
              "#3DAAE9",
            ],
            "data": Array [
              4,
              2,
              7,
              9,
              23,
              2,
            ],
            "label": "My dataset",
          },
        ],
        "labels": Array [
          "Initial Customer Enquiry",
          "J&A",
          "Bid Submitted",
          "Order Received",
          "In Progress",
          "Completed",
        ],
      }
    }
    options={
      Object {
        "cutoutPercentage": 70,
        "legend": Object {
          "display": false,
        },
        "showTotalLabel": true,
      }
    }
  />,
]
`;
